src/renderers/shared/reconciler/__tests__/ReactUpdates-test.js

```diff
@@ -433,10 +433,15 @@ describe('ReactUpdates', function() {
     root = ReactTestUtils.renderIntoDocument(root);
 
     function expectUpdates(desiredWillUpdates, desiredDidUpdates) {
-      expect(willUpdates).toEqual(desiredWillUpdates);
```

Comment:
Yeah, there is also the issue that the lifecycle gets called more often in dev, so the arrays will not be equal anymore due to https://github.com/facebook/react/pull/3516.  We now just verify that every element of desiredWillUpdates dose appear in willUpdates.